### DISCUSSION TITLE
S3client Abnormal operation of gcc12 version in Debian system  （Aws::Client::ClientConfiguration）

### DISCUSSION THREAD

**Discussion by wangdachui0814** [aws/aws-sdk-cpp#3026] [Q&A] [answered]
S3client runs normally under gcc4.8 in Centos system
The gcc12 version runs abnormally in the Debian system, with the exception code line and stack as follows

Aws::Client::ClientConfiguration cfg；
cfg.region = region.c_str();

#0  0x00007ffff54b7e3a in __GI___libc_free (mem=0xbb8) at ./malloc/malloc.c:3362
#1  0x00007ffff4b3fd55 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_mutate(unsigned long, unsigned long, char const*, unsigned long) ()
   from /lib/x86_64-linux-gnu/libstdc++.so.6
#2  0x00007ffff4b40d6e in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned long, unsigned long, char const*, unsigned long) ()
   from /lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff7ab33de in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::assign (__s=0x555555820be0 "cn-beijing", this=0x7fffffffdb38)
    at /usr/include/c++/12/bits/basic_string.h:1647
#4  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::operator= (__s=0x555555820be0 "cn-beijing", this=0x7fffffffdb38) at /usr/include/c++/12/bits/basic_string.h:815
#5  CSOperatorS3::init (this=0x555555720a40) at /home/306838/Cloudstory/PS-StorageClient/Trunk/cpp/S3Plugin/src/CSOperatorS3.cpp:73

(gdb) p cfg
$1 = {userAgent = "", scheme = Aws::Http::Scheme::HTTP, region = <error: Cannot access memory at address 0xbb8>, useDualStack = true, useFIPS = false, maxConnections = 0, httpRequestTimeoutMs = 93824995167016, 
  requestTimeoutMs = 93824995166992, connectTimeoutMs = 140737299715544, enableTcpKeepAlive = false, tcpKeepAliveIntervalMs = 140737299715544, lowSpeedLimit = 0, 
  retryStrategy = std::shared_ptr<Aws::Client::RetryStrategy> (expired, weak count 0) = {
    get() = 0x7ffff4c195d8 <std::basic_string<char, std::char_traits<char>, std::allocator<char> >::_Rep::_S_empty_rep_storage+24>}, endpointOverride = "", proxyScheme = (unknown: 0xf4c195d8), proxyHost = "", 
  proxyPort = 1433758720, proxyUserName = "", proxyPassword = "", proxySSLCertPath = "", proxySSLCertType = "", proxySSLKeyPath = "", proxySSLKeyType = "", proxySSLKeyPassword = "", nonProxyHosts = {
    _vptr.Array = 0x0, m_size = 0, m_data = std::unique_ptr<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >> = {get() = 0x0}}, 
  executor = <error reading variable: Cannot access memory at address 0x9>, verifySSL = false, caPath = "", 
  caFile = <error reading variable: Cannot create a lazy string with address 0x0, and a non-zero length.>, 
  writeRateLimiter = std::shared_ptr<Aws::Utils::RateLimits::RateLimiterInterface> (use count 60, weak count -1) = {get() = 0x3d}, 
  readRateLimiter = std::shared_ptr<Aws::Utils::RateLimits::RateLimiterInterface> (use count -63320, weak count 264275271) = {get() = 0x55555571f660}, httpLibOverride = 60, 
  followRedirects = Aws::Client::FollowRedirectsPolicy::DEFAULT, disableExpectHeader = 199, enableClockSkewAdjustment = 21, enableHostPrefixInjection = 171, enableEndpointDiscovery = {m_storage = {__data = "", 
      __align = {<No data fields>}}, m_value = 0x555555720a40}, profileName = "", useRequestCompression = Aws::Client::UseRequestCompression::FALSE}

**Comment by jmklix** [comment, marked as answer]
Can you provide a dockerfile that reproduces this error on a Debian System?
